### PR TITLE
Expose DateInput module

### PR DESCRIPTION
**Pull request by gregziegan**
#### :thinking: What?

Expose some functions and types from the `DateInput` module.

#### :man_shrugging: Why?

It's currently impossible to use the remote date-based filters with the Table modules as their `DateInput -> msg` arguments require an external msg definition like `FilterDateFieldBy (Maybe DateInput)`.

`type Msg = FilterDateFieldBy (Maybe UI.Internal.DateInput)` is impossible to write since `UI.Internal` is purposely not exposed.

#### :pushpin: Jira Issue

Not tracked.

#### :no_good: Blocked by

Nothing.

#### :clipboard: Pending

- [ ] Add tests, not sure what kinds of tests you all would like
- [ ] Expose more functions?

### :fire: Extra

This is optional, but you can add any other useful information that would not fit in the other topics.

### CODE REVIEW

**Comment by PedroHLC** [comment]
Assigning @kress95 as this filter is used in WMS' tables.

**Comment by gregziegan** [comment]
also, feel free to ignore this PR and make your own. i don't care about contribution credit :) just want to use your table filters in my project!

**Comment by gregziegan** [comment]
also, sorry to expand this any further but there's no issues tab. 

Any ideas on how to solve this? i'd like a column of links / buttons but i'm not sure how to omit one column of filters

filter-issue

**Comment by PedroHLC** [comment]
> Any ideas on how to solve this?

I recommend adding something similar to [`unsortable`](https://package.elm-lang.org/packages/PaackEng/paack-ui/latest/UI-Tables-Stateful#unsortable) adapting it to [`UI.Tables.Stateful#Filters`](https://package.elm-lang.org/packages/PaackEng/paack-ui/latest/UI-Tables-Stateful#filtersEmpty).

As we don't have this use-case in any of the in-house applications, I can't merge this (`unfilterable` proposal) right now, but if you do it in a separate PR, I'll merge it when the need appears.

**Comment by kress95** [comment]
> Assigning @kress95 as this filter is used in WMS' tables.

Seems fine

**Review by kress95 — APPROVED** [review]

File: elm.json
Comment (PedroHLC) [inline elm.json:55]: Re-add the new line at EOF.
(Your editor should be respecting our `.editorconfig`)

File: src/UI/DateInput.elm
Comment (PedroHLC) [inline src/UI/DateInput.elm:28]: Alias to a record is not an opaque type, you can use `{ from : DateInput, to : DateInput }` here.

File: src/UI/DateInput.elm
Comment (PedroHLC) [inline src/UI/DateInput.elm:22]: Alias to a record is not an opaque type, you can use `{ date : DateInput, comparison : PeriodComparison }` here.

**Review by PedroHLC — CHANGES_REQUESTED** [review]
You'll need a rebase for the PR to approve. And don't worry we can do the `elm bump` later for you...

**Comment by PedroHLC** [comment]
@miguelcoba I can't trigger Percy for this PR as it's a branch outside our own repository. Do you have admin powers to force skipping it in this one?

**Comment by PedroHLC** [comment]
I closed this one in 86f2f64de447e3e3632ce1e0e9a2e33750d4f789, note that I moved it to `UI.Utils.DateInput`

**Pull request closed without merging**
